near/omni-types/src/evm/events.rs

```diff
@@ -40,12 +40,12 @@ sol! {
 
 pub fn parse_evm_event<T: SolEvent, V: TryFromLog<Log<T>>>(
     chain_kind: ChainKind,
-    log_rlp: Vec<u8>,
+    log_rlp: &[u8],
 ) -> Result<V, String>
 where
     <V as TryFromLog<Log<T>>>::Error: std::fmt::Display,
 {
-    let rlp_decoded = Log::decode(&mut log_rlp.as_slice()).map_err(stringify)?;
+    let rlp_decoded = Log::decode(&mut log_rlp.to_vec().as_slice()).map_err(stringify)?;
```

Comment:
And extra clone has been introduced here due to `to_vec()`, the `log_rlp` should be moved instead

Reply:
Are you sure? Initially, `log_rlp` was passed by value. Now, it's passed by immutable reference. Is moving it with mutable reference safe if it was not expected earlier?

Reply:
Yes, it looks like Clippy didn't understand the flow correctly. the `log_rlp` should be moved as a `vec` because we need to modify it in the `Log::decode`

Reply:
I would simply return `Vec` here then